handler/actions/src/main/java/io/knotx/fragments/handler/action/http/HttpActionFactory.java

```diff
@@ -36,20 +41,34 @@ public String getName() {
 
   @Override
   public Action create(String alias, JsonObject config, Vertx vertx, Action doAction) {
+    HttpActionOptions options = new HttpActionOptions(config);
+
+    validateNoDoAction(doAction, alias);
+    validateHttpMethodSupported(options, alias);
+
+    WebClient webClient = webClientCache.getOrCreate(vertx, options.getWebClientOptions());
+    return tryToCreateAction(webClient, options, alias);
+  }
+
+  private void validateNoDoAction(Action doAction, String alias) {
     if (doAction != null) {
       throw new ActionConfigurationException(alias, "Http Action can not wrap another action");
     }
+  }
 
-    HttpActionOptions options = new HttpActionOptions(config);
-
-    WebClient webClient = webClientCache.getOrCreate(vertx, options.getWebClientOptions());
+  private void validateHttpMethodSupported(HttpActionOptions options, String alias) {
+    if (!SUPPORTED_METHODS.contains(options.getHttpMethod().toUpperCase())) {
+      throw new ActionConfigurationException(alias,
+          String.format("HttpMethod %s configured for HttpAction is not supported",
+              options.getHttpMethod()));
+    }
+  }
 
-    switch (options.getHttpMethod()) {
-      case "GET":
-        return new HttpAction(webClient, options, alias);
-      default:
-        throw new ActionConfigurationException(alias,
-            "HttpMethod configured for HttpAction is not supported");
+  private Action tryToCreateAction(WebClient webClient, HttpActionOptions options, String alias) {
+    try {
+      return new HttpAction(webClient, options, alias);
+    } catch (IllegalArgumentException cause) {
+      throw new ActionConfigurationException(alias, "Creating HttpAction failed", cause);
```

Comment:
This happens only when both - `body` and `bodyJson` are provided in the options?
I wonder if we can easily cover this case, but it doesn't look like. Well it is not very crucial.

Reply:
I'd prefer that to be a single `body` parameter accepting both JSON and string. If HOCON converter could convert such snippet to string:
```body {
param = value 
}```
then the case could be covered easily - but still, validating the output JSON would be problematic - WDYT?

Reply:
I guess we don't have option for having one field unless writing some custom converter logic. Let's not do this for that case and keep `body` and `bodyJson`. My comment was rather pointing that this exception logic is not covered in unit test.